src/plugins/workspace/server/utils.ts

```diff
@@ -50,3 +54,63 @@ export const getPrincipalsFromRequest = (
 
   throw new Error('UNEXPECTED_AUTHORIZATION_STATUS');
 };
+
+export const updateDashboardAdminStateForRequest = (
+  request: OpenSearchDashboardsRequest,
+  groups: string[],
+  users: string[],
+  configGroups: string[],
+  configUsers: string[]
+) => {
+  if (configGroups.length === 0 && configUsers.length === 0) {
+    updateWorkspaceState(request, {
+      isDashboardAdmin: false,
+    });
+    return;
+  }
+  const groupMatchAny = groups.some((group) => configGroups.includes(group)) || false;
+  const userMatchAny = users.some((user) => configUsers.includes(user)) || false;
+  updateWorkspaceState(request, {
+    isDashboardAdmin: groupMatchAny || userMatchAny,
+  });
+};
+
+export const stringToArray = (adminConfig: string | undefined) => {
+  if (!adminConfig) {
+    return [];
+  }
+  let adminConfigArray;
+  try {
+    adminConfigArray = JSON.parse(adminConfig);
+  } catch (e) {
+    return [];
+  }
+  return adminConfigArray;
+};
+
+export const getApplicationOSDAdminConfig = async (
```

Comment:
```suggestion
export const getOSDAdminConfigFromApplicationConfig = async (
```
What about we changing the function name to like this?

Reply:
Sure, I will rename this function.